crates/rpc-types-eth/src/syncing.rs

```diff
@@ -16,6 +16,9 @@ pub struct SyncInfo {
     pub warp_chunks_amount: Option<U256>,
     /// Warp sync snapshot chunks processed.
     pub warp_chunks_processed: Option<U256>,
+    /// The details of the sync stages as an hashmap
+    /// where the key is the name of the stage and the value is the block number.
+    pub stages: Option<HashMap<String, U256>>,
```

Comment:
let's do a vec here because we expect that these are ordered
```suggestion
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub stages: Option<Vec<String, U64>>,
```

we can also add a helper type for this, so that this is compat with erigon see also:

https://github.com/ledgerwatch/erigon/blob/aefb97b07d1c4fd32a66097a24eddd8f6ccacae0/turbo/jsonrpc/eth_system.go#L55-L60